internal/util/kafka.go

```diff
@@ -22,14 +22,17 @@ type kafkautil struct{}
 
 // NewReader creates a configured kafka.Reader.
 func (k kafkautil) NewReader(topic string) *kafka.Reader {
-	var dialer *kafka.Dialer
-
-	saslMechanism, tlsConfig := getSaslAndTLSConfig()
-	dialer = &kafka.Dialer{
-		Timeout:       10 * time.Second,
-		DualStack:     true,
-		TLS:           tlsConfig,
-		SASLMechanism: saslMechanism,
+	var dialer *kafka.Dialer = kafka.DefaultDialer
+
+	if config.DefaultConfig.KafkaSecurityProtocol != "" {
```

Comment:
I would suggest checking explicitly for `SASL_SSL` here:

```suggestion
	if config.DefaultConfig.KafkaSecurityProtocol == "SASL_SSL" {
```

SecurityProtocol of `SSL` is also valid and it would have no username/password. However, that configuration is not typically in use.

Reply:
Sure @bsquizz, thanks for pointing it out!